miniprofiler/context.go

```diff
@@ -0,0 +1,49 @@
+// +build go1.7
+
+package miniprofiler
+
+import (
+	"context"
+	"net/http"
+)
+
+// contextKey can be used to retreive the profiler instance from the request's context
+const contextKey = "MiniProfiler"
+
+// ContextHandler is an alternate handler that passes the profiler on the http request's context rather than as function arguments.
+// This approach is more compatible with standard net/http Handlers.
+type ContextHandler struct {
+	f http.HandlerFunc
+	p *Profile
+}
+
+// NewContextHandler creates a ContextHandler to wrap the given http.HandlerFunc. A profiler will be added to the request Context, and
+// can be retreived with miniprofiler.GetTimer(r)
+func NewContextHandler(f http.HandlerFunc) http.Handler {
+	return ContextHandler{
+		f: f,
+	}
+}
+
+func (h ContextHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	switch r.URL.Query().Get("pp") {
+	default:
+		h.profileRequest(w, r)
+	}
+}
+
+func (h ContextHandler) profileRequest(w http.ResponseWriter, r *http.Request) {
+	h.p = NewProfile(w, r, FuncName(h.f))
```

Comment:
This happens to work with parallel requests because h is not passed as a pointer. I don't like that design. It doesn't look like `h.p` is used elsewhere, so can it be a a local variable to this function?